diff --git a/src/modules/hub/game_session/infrastructure/trash/repository/repository.go b/src/modules/hub/game_session/infrastructure/trash/repository/repository.go
new file mode 100644
index 0000000..a07478e
--- /dev/null
+++ b/src/modules/hub/game_session/infrastructure/trash/repository/repository.go
@@ -0,0 +1,78 @@
+package trash_repo
+
+import (
+	"context"
+	"errors"
+
+	"github.com/root9464/Go_GamlerDefi/src/packages/lib/logger"
+	"gorm.io/gorm"
+)
+
+type TrashRepository struct {
+	logger *logger.Logger
+	db     *gorm.DB
+}
+
+type Ticket struct {
+	GameID        string `json:"game_id"`
+	GameSessionID uint   `json:"game_session_id"`
+	PlayerID      string `json:"player_id"`
+	Code          string `json:"code"`
+}
+
+type Session struct {
+	HostID        string `json:"host_id"`
+	GameSessionID uint   `json:"game_session_id"`
+}
+
+func NewTrashTicketRepository(logger *logger.Logger, db *gorm.DB) *TrashRepository {
+	return &TrashRepository{
+		logger: logger,
+		db:     db,
+	}
+}
+
+func (r *TrashRepository) IsUserHasAccess(ctx context.Context, sessionID uint, userID string) (bool, error) {
+	r.logger.Infof("check access for session: %v and user: %s", sessionID, userID)
+
+	var ticket Ticket
+	err := r.db.WithContext(ctx).
+		Where(&Ticket{
+			GameSessionID: sessionID,
+			PlayerID:      userID,
+		}).
+		First(&ticket).Error
+
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			r.logger.Warnf("Access denied: No ticket found for user %s in session %v", userID, sessionID)
+			return false, nil
+		}
+		r.logger.Errorf("Database error while checking access for user %s: %v", userID, err)
+		return false, err
+	}
+
+	r.logger.Infof("Access granted: Ticket found for user %s in session %v", userID, sessionID)
+	return true, nil
+}
+
+func (r *TrashRepository) GetSessionHost(ctx context.Context, sessionID uint) (string, error) {
+	var session Session
+	err := r.db.WithContext(ctx).
+		Where(&Session{
+			GameSessionID: sessionID,
+		}).
+		First(&session).Error
+
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			r.logger.Warnf("Session host not found for session %v", sessionID)
+			return "", nil
+		}
+		r.logger.Errorf("Database error while getting session host for session %v: %v", sessionID, err)
+		return "", err
+	}
+
+	r.logger.Infof("Session host found for session %v: %s", sessionID, session.HostID)
+	return session.HostID, nil
+}
diff --git a/src/packages/utils/request.go b/src/packages/utils/request.go
new file mode 100644
index 0000000..e1d9f7d
--- /dev/null
+++ b/src/packages/utils/request.go
@@ -0,0 +1,114 @@
+package utils
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+type APIClient struct {
+	BaseURL    *url.URL
+	HTTPClient *http.Client
+	Headers    http.Header
+}
+
+type RequestConfig struct {
+	Ctx    context.Context
+	Method string
+	Path   string
+
+	// Необязательные поля
+	Headers     http.Header
+	QueryParams url.Values
+	Body        any
+	Result      any
+	ErrorResult any
+}
+
+func NewAPIClient(baseURL string) (*APIClient, error) {
+	parsedBaseURL, err := url.Parse(baseURL)
+	if err != nil {
+		return nil, fmt.Errorf("некорректный базовый URL: %w", err)
+	}
+
+	return &APIClient{
+		BaseURL: parsedBaseURL,
+		HTTPClient: &http.Client{
+			Timeout: time.Second * 15,
+		},
+		Headers: make(http.Header),
+	}, nil
+}
+
+func (c *APIClient) Do(config *RequestConfig) (*http.Response, error) {
+	reqURL, err := c.BaseURL.Parse(config.Path)
+	if err != nil {
+		return nil, fmt.Errorf("не удалось сформировать URL запроса: %w", err)
+	}
+	if config.QueryParams != nil {
+		reqURL.RawQuery = config.QueryParams.Encode()
+	}
+
+	var reqBody io.Reader
+	if config.Body != nil {
+		bodyBytes, err := json.Marshal(config.Body)
+		if err != nil {
+			return nil, fmt.Errorf("не удалось сериализовать тело запроса в JSON: %w", err)
+		}
+		reqBody = bytes.NewBuffer(bodyBytes)
+	}
+
+	req, err := http.NewRequestWithContext(config.Ctx, config.Method, reqURL.String(), reqBody)
+	if err != nil {
+		return nil, fmt.Errorf("не удалось создать HTTP-запрос: %w", err)
+	}
+
+	for key, values := range c.Headers {
+		for _, value := range values {
+			req.Header.Add(key, value)
+		}
+	}
+	for key, values := range config.Headers {
+		for _, value := range values {
+			req.Header.Add(key, value)
+		}
+	}
+	if config.Body != nil {
+		if req.Header.Get("Content-Type") == "" {
+			req.Header.Set("Content-Type", "application/json; charset=utf-8")
+		}
+	}
+
+	resp, err := c.HTTPClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("ошибка при выполнении HTTP-запроса: %w", err)
+	}
+	defer resp.Body.Close()
+
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return resp, fmt.Errorf("не удалось прочитать тело ответа: %w", err)
+	}
+
+	if resp.StatusCode >= 200 && resp.StatusCode <= 299 {
+		if config.Result != nil {
+			if err := json.Unmarshal(respBody, config.Result); err != nil {
+				return resp, fmt.Errorf("не удалось десериализовать успешный ответ: %w", err)
+			}
+		}
+	} else {
+		if config.ErrorResult != nil {
+			if err := json.Unmarshal(respBody, config.ErrorResult); err != nil {
+				return resp, fmt.Errorf("API вернул ошибку (статус %d): %s", resp.StatusCode, string(respBody))
+			}
+		}
+		return resp, fmt.Errorf("API вернул ошибку (статус %d): %s", resp.StatusCode, string(respBody))
+	}
+
+	return resp, nil
+}
diff --git a/tmp/cert/cert.pem b/tmp/cert/cert.pem
new file mode 100644
index 0000000..f0d63b0
--- /dev/null
+++ b/tmp/cert/cert.pem
@@ -0,0 +1,29 @@
+-----BEGIN CERTIFICATE-----
+MIIFCTCCAvGgAwIBAgIUGe9DrpvQMPAHup+NQGCT5ehxmj4wDQYJKoZIhvcNAQEL
+BQAwFDESMBAGA1UEAwwJbG9jYWxob3N0MB4XDTI1MDgyOTE2MzQzN1oXDTI2MDgy
+OTE2MzQzN1owFDESMBAGA1UEAwwJbG9jYWxob3N0MIICIjANBgkqhkiG9w0BAQEF
+AAOCAg8AMIICCgKCAgEAz5NBgTbkAxTFlGOjNbPi/QaPLUUlzlQ792QULjnLPi3/
+KzQP2FwQ0hEq+Ge0Yu6X4sZhV2ASft4UJw8/7ERPX6UIPpI04KJ2Yo/n3XUTw0nU
+lrQJb+nTrQe4C6DDiapHEWl9G8jsnLWhrhXI9ObmhzXDEL7b/foluNc+bOfr2/5q
+aCEzVo7hwAVyLIew74PM4poGkq2wFkakU9oGl8kCa14sTydvoaITm7dKg3R3+1ga
+SQ8HuOvSUOm22ZEZ8baAeAxmdhImPO6jg4TLCicrSm9mSQWvmKqQ7/i1BaHIMucD
+xW8yO2BYVGIK97icDhac7XnZYZpZ9A/ikw7SopdkxsWCzkwHd+54kwSyzFqJpHe6
+jsRNvX0iw8AaNKP5xx2eNUGteOmffPnDzr0DQG9KR0DZm+Rw2xeLhBEuLsHStqpN
+uStzOSonwfbyRiOJK/0Eofw+147b2nsOoJvOupKdDkOUG/OGxuceRH1CjPV6NngF
+yj4uzXYyB9hTkhTEH7gQEbtHYMBGUKCXDdChwfKZLNysL+ZNRmyhgqGCJXA35/61
+IB7Vu/Nddz+HeG2HgKUr1iAR1OWKF39NCSW5piBTXf4T20QE3Wb1GQwH/69ANu+a
+ylajODXC8V2zEa1OKkZ06Rc2xDOfHcOU2ukxkYkoQNiZd0o+7pTEm/2aAr0jjXsC
+AwEAAaNTMFEwHQYDVR0OBBYEFE98P/zdLw4g2oZ51BQBZNHsQl5IMB8GA1UdIwQY
+MBaAFE98P/zdLw4g2oZ51BQBZNHsQl5IMA8GA1UdEwEB/wQFMAMBAf8wDQYJKoZI
+hvcNAQELBQADggIBAJLRJcrou7dk0P7eVvnJCsoZV6+q5kfGdvCn+MDstL42cX9p
+vj3H0XKYwFjgU1eFeuO+Od93GoedbOERaClISm1N3jjVgECccEpBQYOBxJhPbQZu
+zmBsvv1aLdHfBbZfpyqCFag92R1zIPWBZu1qzT5HikmHCt7EubT+3u5dw9gCLc+Y
+A9J9vCZCcbCSXn+nzftDK+nVEXlMYeIsFiNY4FAI0u7IINtkw4PG4+P/j9XBekQn
+EM4v+IHJTLc+au1mR6+v58DTzgGMxR2SCyPTSd1i6pBJtPhMSI4Sfua66OBFDcc6
+y7uhLfWfP50sPQowx+FOqp2FVgbRPeK9fXXnOyTSroCO3YcznwKCuZAWJizLl/XV
+XmTdJeWFMSaeC+jiZZSytjEpqyXfiA1yIoXkLt2/gEbdq3Ng9fW/6qjAH0Koq5Kb
+/yIWAJo589UYW3aoCJSBqgJx1Aw4olIDIgGiuUmRHsGjHKT5NpwUU7Bw77vZe2w7
+bSCNY/QVuiOyQdHyEVibVCiNjpKaw6OhhZ40EWmF5Dc8jgoY8GBo5EFGPej37ngM
+1u+aHCMGQG6hCVO9fnOU3Ka+hktHJgqqxNYoefqAolZEqrcbJDzeLQusk8a9YDwR
+O4CDt74Jay+tgMBmGI+HhHs/f9urMix17t6twKdMo5+WwQYBGWVl0Anjf6gE
+-----END CERTIFICATE-----
diff --git a/tmp/cert/key.pem b/tmp/cert/key.pem
new file mode 100644
index 0000000..18bd562
--- /dev/null
+++ b/tmp/cert/key.pem
@@ -0,0 +1,52 @@
+-----BEGIN PRIVATE KEY-----
+MIIJQgIBADANBgkqhkiG9w0BAQEFAASCCSwwggkoAgEAAoICAQDPk0GBNuQDFMWU
+Y6M1s+L9Bo8tRSXOVDv3ZBQuOcs+Lf8rNA/YXBDSESr4Z7Ri7pfixmFXYBJ+3hQn
+Dz/sRE9fpQg+kjTgonZij+fddRPDSdSWtAlv6dOtB7gLoMOJqkcRaX0byOyctaGu
+Fcj05uaHNcMQvtv9+iW41z5s5+vb/mpoITNWjuHABXIsh7Dvg8zimgaSrbAWRqRT
+2gaXyQJrXixPJ2+hohObt0qDdHf7WBpJDwe469JQ6bbZkRnxtoB4DGZ2EiY87qOD
+hMsKJytKb2ZJBa+YqpDv+LUFocgy5wPFbzI7YFhUYgr3uJwOFpztedlhmln0D+KT
+DtKil2TGxYLOTAd37niTBLLMWomkd7qOxE29fSLDwBo0o/nHHZ41Qa146Z98+cPO
+vQNAb0pHQNmb5HDbF4uEES4uwdK2qk25K3M5KifB9vJGI4kr/QSh/D7Xjtvaew6g
+m866kp0OQ5Qb84bG5x5EfUKM9Xo2eAXKPi7NdjIH2FOSFMQfuBARu0dgwEZQoJcN
+0KHB8pks3Kwv5k1GbKGCoYIlcDfn/rUgHtW78113P4d4bYeApSvWIBHU5YoXf00J
+JbmmIFNd/hPbRATdZvUZDAf/r0A275rKVqM4NcLxXbMRrU4qRnTpFzbEM58dw5Ta
+6TGRiShA2Jl3Sj7ulMSb/ZoCvSONewIDAQABAoICACfZ14udJC1LQZt5fY4D2SJG
+rho82kNAfLrEJfOx+MkZP5Hlztz03+3UifFix4ju0Y4iE/IYabSdojTVgqozeyGb
+p3eLzkcKKbxGcHdd0Mt/UycTrXQk6hSgFL92v9OgM0CuhexdqL91R3MKj6gk5EE1
+GxwN6GJcMf3IpR13cRqtP3R6W0bU65ZX9VGDQ37YSO+ALp7eZZ4BNe87PmTJnINM
+cylHIvFfEnCkwdN2+snwYiuAjTT9YUVTeV1PtDFpnkRt7QXh+iFcl1valLiASJlj
+xbljNBEVPfo0P1yzcKIFI/41mOk5V/vrTru8X4l84efdqtRIM9jda9xbziiU9C20
+Ud7aotyf4w9/y6LwHsLQ6SntRgQOgt8kAxsleozL/Bck8zHEvasBvhpkzlq6jWo7
+c2/y7wsnZw7/bFiPmJTbp9czGbFzuaDOx77Ubmfx2swoceeypA2CGNnKRKwhvFjX
+MIL1gOxfW9a7ifZytWGzvXmFfPYnxamC/Fzh7l+QbR08hvGlgUj+uvbg8G7h3ndS
+XZRnNBPTxg/gsj3XaF8W4ktaTew6BoUxxJ8n55w6EOBiGddUeRgNVm6XFpLVF0Gc
+pVZJKnAmpxxkDQsZPpsoY6fvsY1Lsh6/BA5dUOKWnTMa7RS+cPQuYMbH2r2KTh13
+kQs/8hVR4/nTihTAmzwhAoIBAQD6Px/HIRwFl508+LsXO/9TxK6qqFt5bBTHjND0
+wb3CUGTQA56zbw2al9mjTQGahcNWrLrKu9w9DkAwGElPPg4f5q1J9MWvYg/0hVN8
+V5ZdfaWVqKSwFHvMeeqh+P8lu2Z5OYOl6FrmgAVjw52J9r+pQ5uZ2HmE3WXvTkhO
+B9bvemux6MN66AqlgtgP16d3K5Mu8IlvkpB1iE2TzUTKJtQJZZPOMBX17h6HPME0
+I8BoXTykLX++mN6YPygLWsqtGw7xIh7hn98F9MshjjOHf2WPE5y/V5yQjqiap5/Q
+GPiVawhRO4/jpACTtD1JlZ74yxFPReXXCB9yxkau9+yKrGYzAoIBAQDUWPsiRAI/
+ZgnyTzFLFPDLqCzhCww5CxaESYfO1/Rji+33Suaf1v8xaP6F0XAXisGBpD6GLKwI
+tllvzzZ3+DmbRqMokQldKppwnHHQt1W60TW2Qy6WGydghjn2H1Z1IeXlwOTti4CL
+8mXaH2Nhq1W9+BMnj/9JRaGcuUa35c0utdf5agupfrIPq6MEEa+oH1LGQKkC8qSN
+N6cZyjr+kdyCPkoYYiC4pgNhdOVHK0Qrxl7TS8kzyutfni2oCkILScB2zuyi1mtK
+wdG83tCgK4HC3dkA3RDGdlxaUz1sbAJ1BUkRLaE3fWaHjSvfoQvk+MzxkldHkGO0
+IvZ2JeK02KOZAoIBAA5tCSXvkK/514SmShLC0KtBkVghMHTFeEUwpvMVIB1NiSI8
+sy9g3YXl7xPs4oTH5UTf3DzHq6CRa73aCI/a4fGQ2LiEtciwfH76GJ/4ZKCrF5qT
+CDtmUuSsuovv0LWaIgQm/qHXzTWwd58SrchAQzg8v4GU6xxvruqmtqBY71kK056D
+7pDtQ3/bFR2ErxC9oFR6ZVHbPxynaOF2WdrVdi+8fVLLjtiIi5X8OJ8K2JKrnU1R
+miWgpmRcb7CH4VstTiU0dFdJYg/z1qnuiqXrAC3JUaXuJQZoUuvncd0LM3lZnDci
+md7jm2zoXQga6fkpq6vIOcG1pCp3p6d6f1ZXLeUCggEANAlKIW+BDYA318VIUxni
+mEWyGcPXTvNEpU5PlYSeQva7bepQ8cKB7t3H4YsdaOd73AUfvUmdLtVgvWGN+4A8
+vaxltz8ziD00c3wi9/61Z351JyD3fZlR4PmDoqizSyjvaYoHj56r9Z+FDhq7A+g0
+rsToDL3g1VfW7s0IY3BRA61UK66tBPAjOmvZGtoqrYxfgG2//RTw9ho5PLyuEEw2
+OUSCa89ALX+hPHSk6vfTUNbfze+4cpS7cUcV8SPyY0nDfFng61aYHE261AnWHioL
+NnGfB/MkdT30C75yQuuE/qA485i29R4pzfXuVnLks0u1DU3LTaI4jKA0vWczUxXz
+oQKCAQEAo4VY3ezacIjSMLxHTBGiy84mOHc15jLvzoUZGR2PkWWtBVUv3KsU1nBs
+IHO2mJOgIuiLNgCipWEaIpHjMG/WbPSpQfuZ8kkZYURnH9zeRZqZMZJcssO2ETJP
+LgcWgzpEwMGANjDIvWgzMYcaclbJCfDX4kqzdMEtuQkPs1KsvAkJM1GKW0+igDYe
+2pE2HOlViOM7BeHIg7InddJ9qoR8r6jLfkbbeKnUmrLclSM6KI/haOwFmoNLUaQS
+htMw4H6tEYHPWQTB/e25l0l9d1JmLVGA7zU06+y6hEAVjWLSbf/v7xRghkH5fi0k
++4d+dKNAxpK6ziWz5rrs/1mIJBOiIQ==
+-----END PRIVATE KEY-----
